x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/action/EqlSearchRequest.java

```diff
@@ -0,0 +1,242 @@
+import org.elasticsearch.index.query.AbstractQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+public class EqlSearchRequest extends ActionRequest implements CompositeIndicesRequest, ToXContent {
+
+    private String index = "";
+    private QueryBuilder query = null;
+    private String timestampField = "@timestamp";
+    private String eventTypeField = "event.category";
+    private String implicitJoinKeyField = "agent.id";
+    private int fetchSize = 50;
+    private List<String> searchAfter = Collections.emptyList();
+    private String rule;
+
+    static final String KEY_QUERY = "query";
+    static final String KEY_TIMESTAMP_FIELD = "timestamp_field";
+    static final String KEY_EVENT_TYPE_FIELD = "event_type_field";
+    static final String KEY_IMPLICIT_JOIN_KEY_FIELD = "implicit_join_key_field";
+    static final String KEY_SIZE = "size";
+    static final String KEY_SEARCH_AFTER = "search_after";
+    static final String KEY_RULE = "rule";
+
+    static final ParseField QUERY = new ParseField(KEY_QUERY);
+    static final ParseField TIMESTAMP_FIELD = new ParseField(KEY_TIMESTAMP_FIELD);
+    static final ParseField EVENT_TYPE_FIELD = new ParseField(KEY_EVENT_TYPE_FIELD);
+    static final ParseField IMPLICIT_JOIN_KEY_FIELD = new ParseField(KEY_IMPLICIT_JOIN_KEY_FIELD);
+    static final ParseField SIZE = new ParseField(KEY_SIZE);
+    static final ParseField SEARCH_AFTER = new ParseField(KEY_SEARCH_AFTER);
+    static final ParseField RULE = new ParseField(KEY_RULE);
+
+    private static final ObjectParser<EqlSearchRequest, Void> PARSER = objectParser(EqlSearchRequest::new);
+
+    public EqlSearchRequest() {
+        super();
+    }
+
+    public EqlSearchRequest(StreamInput in) throws IOException {
+        super(in);
+        index = in.readString();
+        query = in.readOptionalNamedWriteable(QueryBuilder.class);
+        timestampField = in.readString();
+        eventTypeField = in.readString();
+        implicitJoinKeyField = in.readString();
+        fetchSize = in.readVInt();
+        searchAfter = in.readList(StreamInput::readString);
+        rule = in.readString();
+    }
+
+    public EqlSearchRequest(String index, QueryBuilder query,
+                            String timestampField, String eventTypeField, String implicitJoinKeyField,
+                            int fetchSize, List<String> searchAfter, String rule) {
+        this.index = index;
+        this.query = query;
+        this.timestampField = timestampField;
+        this.eventTypeField = eventTypeField;
+        this.implicitJoinKeyField = implicitJoinKeyField;
+        this.fetchSize = fetchSize;
+        this.searchAfter = searchAfter;
+        this.rule = rule;
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        return null;
```

Comment:
Since we rely on the fact that index, timestampField, eventTypeField, etc cannot be null and fetchSize cannot be negative, we should validate that it's indeed the case. We could also make serialization more forgiving and avoid serializing the same default values on every request.

Reply:
Will add validation code